File: app/src/main/java/com/example/borysgurtovyi/codinginterviewprep/MoveZeroes.java
package com.example.borysgurtovyi.codinginterviewprep;

/**
 * Created by borysgurtovyi on 2/12/18.
 */

public class MoveZeroes {


    //   [0, 1, 0, 3, 12]

    //   [1, 3, 12, 3, 12]

    //   [1, 3, 12, 0, 0]



    // 1, 2, 50, 7, 10

    public void moveZeroes(int[] nums) {

        int k = 0;

        for(int i = 0; i < nums.length; i++) {
            if(nums[i] != 0 ) {
                nums[k] = nums[i];
                k++;
            }
        }

        for( ; k < nums.length ; k++ ) {
            nums[k] = 0;
        }
    }
}

File: app/src/main/java/com/example/borysgurtovyi/codinginterviewprep/MoveZeroes.kt
package com.example.borysgurtovyi.codinginterviewprep

/**
 * Created by lisa on 15.02.2018.
 */
class MoveZeroes{

    fun moveZeroes(nums: Array<Int>) {
        var i = 0
        var k = 0
        for (i in 0..nums.size - 1){
            if (nums[i] != 0){
                nums[k] = nums[i]
                k += 1
            }
        }
        for (k in k..nums.size - 1){
            nums[k] = 0
        }
    }
}